### PR TITLE
Add check for violation of guiderates

### PR DESCRIPTION
**Pull request by totto82**

### CODE REVIEW

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
I have manually checked the failures and the results are fine.

File: opm/simulators/wells/WellGroupHelpers.cpp [hunk 1]
```diff
@@ -1181,6 +1186,19 @@ namespace WellGroupHelpers
             }
         }
 
+        // check whether guide rate is violated
+        if(local_reduction_level > 0) {
```
Comment (atgeirr) [inline opm/simulators/wells/WellGroupHelpers.cpp:1190]: Nitpicks:
Space after if.

Below, I think adding an alias
```
const auto& guided_group = chain[local_reduction_level];
```
or similar would improve readability. All variables here can also be const.

(Same for injector function)

**Review by atgeirr — APPROVED** [review]
Looks good to me, I only have some minor nitpicks.

**Comment by totto82** [comment]
jenkins build this please

File: opm/simulators/wells/WellGroupHelpers.cpp (hunk 1 above)
Comment (totto82) [reply opm/simulators/wells/WellGroupHelpers.cpp]: Thanks for the suggestions.

**Comment by totto82** [comment]
jenkins build this update_data please

**Comment by totto82** [comment]
jenkins build this opm-tests=728 please

**Pull request merged**